projects/storefrontlib/src/cms-structure/page/slot/page-slot.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PageSlotService {
+  protected instantSsrSlots: string[] | undefined;
+
+  constructor(
+    @Inject(PLATFORM_ID) platformId: any,
+    @Inject(DOCUMENT) document
+  ) {
+    if (isPlatformBrowser(platformId)) {
+      // Find all slots visible in the SSR pre-rendered DOM
+      this.instantSsrSlots = Array.from(
+        document.querySelectorAll('cx-page-slot')
+      )
+        .filter(
+          (el: Element) =>
+            el.getBoundingClientRect().top <
+            document.documentElement.clientHeight
+        )
+        .map((el: Element) => el.classList[0]);
+    }
+  }
+
+  shouldNotDefer(slot: string): boolean {
```

Comment:
what about `isPrerendered()`?

Reply:
actually.. going forward, we probably want to move the component logic to this service. We can take a first stab at it by renaming this method, and implement all the logic of `PageSlotComponent.getComponentDeferOptions()`. That will make this API better readable (we hide the SSR complexity from the API).

Reply:
👍

Reply:
Is this method still intended to be part of public API?

Reply:
nit: what about renaming, I don't like the `Not` part of it. Either `shouldDefer()` or `isPrerendered`.